pkg/deps/jar.go

```diff
@@ -0,0 +1,136 @@
+//
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package deps
+
+import (
+	"archive/zip"
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"path/filepath"
+	"regexp"
+
+	"github.com/apache/skywalking-eyes/license-eye/internal/logger"
+	"github.com/apache/skywalking-eyes/license-eye/pkg/license"
+)
+
+type JarResolver struct{}
+
+func (resolver *JarResolver) CanResolve(jarFile string) bool {
+	return filepath.Ext(jarFile) == ".jar"
+}
+
+func (resolver *JarResolver) Resolve(jarFile string, report *Report) error {
+	state := NotFound
+	if err := resolver.ResolveJar(&state, jarFile, report); err != nil {
```

Comment:
In this resolver (jar resolver), we should support a path pattern like `dist/**/*.jar` and expand all the `.jar` files and resolve their licenses, the current feature in this PR only support a singe `.jar` file, right?

Reply:
From my understanding, JarResolver is only responsible for resolve the licenses of jar files. And `deps/config.go` will enter all jar files in the `dist/**/*.jar`

Reply:
> From my understanding, JarResolver is only responsible for resolve the licenses of jar files. And `deps/config.go` will enter all jar files in the `dist/**/*.jar`

Well, now, it seems `deps/config.go` doesn't expand the pattern `dist/**/*.jar` to all `.jar` files (like `dist/module1/log4j.jar`, `dist/module2/junit.jar`), we should do this, right?

Reply:
>
> Well, now, it seems `deps/config.go` doesn't expand the pattern `dist/**/*.jar` to all `.jar` files (like `dist/module1/log4j.jar`, `dist/module2/junit.jar`), we should do this, right?

Yes, it will be complete in the next pr

Reply:
> > Well, now, it seems `deps/config.go` doesn't expand the pattern `dist/**/*.jar` to all `.jar` files (like `dist/module1/log4j.jar`, `dist/module2/junit.jar`), we should do this, right?
> 
> Yes, it will be complete in the next pr

Nice